src/unit_tests/cache_tests.cpp

```diff
@@ -167,3 +167,8 @@ BOOST_AUTO_TEST_CASE(passive_multiple_test)
     cache.get(uri_options1, 16);
     BOOST_TEST(cache.count(uri_options1) <= 8);
 }
+
+BOOST_AUTO_TEST_CASE(destroy)
+{
+    GDALDestroyDriverManager();
+}
```

Comment:
We were lucky or that's the GDAL behavior change; without it tests fail. 

Add it in all cases we wan't to undo `GDALAllRegister()` calls.

```bash
github-build-linux-amd64-1  | corrupted size vs. prev_size
github-build-linux-amd64-1  | Aborted
github-build-linux-amd64-1  | make[1]: *** [Makefile:16: tests] Error 134
github-build-linux-amd64-1  | make[1]: Leaving directory '/workdir/src/unit_tests'
github-build-linux-amd64-1  | make: *** [Makefile:39: tests] Error 2
github-build-linux-amd64-1  | make: Leaving directory '/workdir/src'
```